### PR TITLE
refactor(pages): rebuild revocation failure as react page component

### PR DESCRIPTION
**Pull request by punknoir101**
# Short Description
- rebuild revocation failure in react
- add unit and integration tests

# Changes
- adds new page for revocation failure in client
- adds story for revocation failure in client
- replace template part with react component template in unlock_code_revocation_steps
- fix some small code style issues
- add constants in unlock_code_revocation_steps to avoid string repetitions

# Feedback
- all 

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

File: webapp/app/templates/react_template.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from flask import render_template
+
+REACT_COMPONENT = 'react_component.html'
```
Comment (JulianRoesner) [inline webapp/app/templates/react_template.py:3]: Why do we still need that constant? Couldn't we put that directly in line 6?

File: webapp/app/templates/react_template.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+def render_react_template(component, props=None, form=None, header_title=None):
```
Comment (JulianRoesner) [inline webapp/app/templates/react_template.py:5]: I like that a lot. Clever way 👍

File: webapp/app/templates/react_template.py (hunk 1 above)
Comment (SannyNguyenHung) [reply webapp/app/templates/react_template.py]: Done!

File: webapp/app/templates/react_template.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from flask import render_template
+
+REACT_COMPONENT = 'react_component.html'
+
+def render_react_template(component, props=None, form=None, header_title=None):                          
+    return render_template(REACT_COMPONENT,
+            component=component,
+            props=props,
+            form=form,
+            header_title=header_title)
```
[the remark this answers is not in the document]
Comment (SannyNguyenHung) [reply webapp/app/templates/react_template.py]: Done!

**Review by SannyNguyenHung — APPROVED** [review]
LGTM! Lets merge! :rocket:

**Pull request merged**